components/UI/ReasoningFilterMenu/index.tsx

```diff
@@ -99,9 +114,9 @@ const ReasoningFilterMenu = ({ onChange }: ReasoningFilterMenuProps) => {
           disableTouchRipple
         >
           {
-            reasoningFilterMenuOptions.find(
-              (option) => option.value === selectedValue
```

Comment:
what is the gain here?

Reply:
> I think we should be using the URL as our source of truth here. Put the parameters to the URL as a query param, update the URL and listen for the URL params in the relevant components without extra state in between. We can put the logic into a `useMapOptions` hook and use that in our components. I can take this or help with if needed.
> 
> This is important because usability and ease of sharing is crucial. People can change the options and share the link with a friend or someone on the ground.

I aggree there are a lot of improvements can be done here, for example there are 2 states which can be just one, but I honestly didn't touch the parts of outside of the feature. So I didn't refactor anything.